### PR TITLE
ncm-autofs: Add direct mount support

### LINKED ISSUE #335: ncm-autofs : Not doing the right things when trying to configure a direct mount
I've tried this setting : `/software/components/autofs/maps/kalitest/type = "direct"`
for one of my autofs shares. But then, ncm-autofs adds the following line to /etc/auto.master:
`/kalitest      direct:/etc/auto.direct`
That's not correct. According to "man 5 auto.master", the right way should be :
`/-    /etc/auto.direct`
There's also a problem in the file /etc/auto.direct. It says :
`kalitest    myfileserver:/mydirectory`
while it should say :
`/kalitest    myfileserver:/mydirectory`

### PR DESCRIPTION
**Pull request by samary**
Slight modification to add the direct mount support.

Fixes #335 

This fix works with the following pan example:
```
include {'features/nfs/client/autofs'};
include {'components/autofs/config'};

variable USER_ENTRIES = {
    tbl = nlist(escape("/MY_MOUNTPOINT"),nlist("location","MY_SERVER:/MY_EXPORT","options",""));
};

"/software/components/autofs/maps/MY_MOUNTPOINT" = nlist(
    "enabled",true,
    "mapname","/etc/auto.direct",
    "mountpoint","/-",
    "options","-rw",
    "type","direct",
    "entries", USER_ENTRIES
);
```

Credit @StephaneGerardVUB

### CODE REVIEW

**Comment by jrha** [comment]
please test this

**Comment by jrha** [comment]
Nice!

File: ncm-autofs/src/main/perl/autofs.pm [hunk 1]
```diff
@@ -274,8 +275,9 @@ sub Configure($$@) {
       foreach my $map (keys(%mount_points)) {
         my $map_attrs = $master_entry_attrs{$map};
         foreach my $mountp ( @{$mount_points{$map}} ) {
+          my $map_type_prefix = $map_attrs->{type} eq 'direct' ? '' : $map_attrs->{type}.':';
           $master_contents .= $map_attrs->{prefix}.
-                                       "$mountp\t".$map_attrs->{type}.":$map\t".$map_attrs->{options}."\n";
+                                       "$mountp\t".$map_type_prefix."$map\t".$map_attrs->{options}."\n";
```
Comment (stdweird) [inline ncm-autofs/src/main/perl/autofs.pm]: make this 
```perl
"$mountp\t$map_type_prefix$map\t".$map_attrs->{options}."\n"
```

File: ncm-autofs/src/main/perl/autofs.pm [hunk 2]
```diff
@@ -257,12 +257,13 @@ sub Configure($$@) {
...
         my $map_attrs = $master_entry_attrs{$map};
+        my $map_type_prefix = $map_attrs->{type} eq 'direct' ? '' : $map_attrs->{type}.':';
         foreach my $mountp ( @{$mount_points{$map}} ) {
           $self->debug(2,"Checking entry for mount point $mountp (map $map)...");
           $cnt += $self->updateMap($master_contents_ref,
                                    '^#?\s*('.$error_prefix.'\s*)?'.$mountp.'\s+.*',
-                                   '^'.$map_attrs->{prefix}.$mountp.'\s+'.$map_attrs->{type}.':'.$map.'\s+'.$map_attrs->{options}.'\s*$',
-                                   $map_attrs->{prefix}."$mountp\t".$map_attrs->{type}.":$map\t".$map_attrs->{options},
+                                   '^'.$map_attrs->{prefix}.$mountp.'\s+'.$map_type_prefix.$map.'\s+'.$map_attrs->{options}.'\s*$',
+                                   $map_attrs->{prefix}."$mountp\t".$map_type_prefix."$map\t".$map_attrs->{options},
```
Comment (stdweird) [inline ncm-autofs/src/main/perl/autofs.pm]: move `$map_type_prefix` inside double quotes

**Comment by jrha** [comment]
@samary are you able to update this to address @stdweird's comments?

**Comment by samary** [comment]
@stdweird @jrha : Done

**Comment by jrha** [comment]
Lovely, thanks.

**Pull request merged**